client/my-sites/plugins/plugin-upload/index.jsx

```diff
@@ -121,70 +115,40 @@ class PluginUpload extends Component {
 		);
 	}
 
-	setOpenModal = ( isOpen ) => {
-		this.setState( { showTrialAcknowledgeModal: isOpen } );
-	};
-
-	trialRequested = () => {
-		this.setState( { hasRequestedTrial: true, showEligibility: false } );
-	};
-
 	requestUpdatedSiteData = ( isTransferring, wasTransferring, isTransferCompleted ) => {
 		if ( isTransferring ) {
 			this.setState( { isTransferring: true } );
 		}
 		if ( wasTransferring && isTransferCompleted ) {
-			this.props.fetchUpdatedData();
 			this.setState( { isTransferring: false } );
 		}
 	};
 
 	render() {
-		const {
-			translate,
-			isJetpackMultisite,
-			siteId,
-			siteSlug,
-			isEligibleForHostingTrial,
-			isJetpack,
-			isTrialSite,
-			isAtomic,
-		} = this.props;
-		const { showEligibility, showTrialAcknowledgeModal, isTransferring, hasRequestedTrial } =
-			this.state;
-
-		const showEligibilityWarnings =
-			showEligibility && ! isTransferring && ! isTrialSite && ! hasRequestedTrial;
+		const { translate, isJetpackMultisite, siteId, siteSlug, isJetpack, isTrialSite, isAtomic } =
+			this.props;
+		const { showEligibility, isTransferring } = this.state;
+
+		const showEligibilityWarnings = showEligibility && ! isTransferring && ! isTrialSite;
 
 		return (
 			<Main>
 				<PageViewTracker path="/plugins/upload/:site" title="Plugins > Upload" />
 				<QueryEligibility siteId={ siteId } />
 				<NavigationHeader navigationItems={ [] } title={ translate( 'Plugins' ) } />
 				<HeaderCake onClick={ this.back }>{ translate( 'Install plugin' ) }</HeaderCake>
-				{ ! showTrialAcknowledgeModal && ! isJetpack && (
-					<HostingActivateStatus
-						context="plugin"
-						onTick={ this.requestUpdatedSiteData }
-						keepAlive={ hasRequestedTrial && ! isJetpack && ! isAtomic }
-					/>
+				{ ! isJetpack && (
+					<HostingActivateStatus context="plugin" onTick={ this.requestUpdatedSiteData } />
 				) }
 				{ isJetpackMultisite && this.renderNotAvailableForMultisite() }
 				{ showEligibilityWarnings && (
 					<EligibilityWarnings
 						backUrl={ `/plugins/${ siteSlug }` }
 						onProceed={ this.onProceedClick }
-						showFreeTrial={ isEligibleForHostingTrial }
 					/>
 				) }
 				{ ( ( ! isJetpackMultisite && ! showEligibility ) || isAtomic || isTrialSite ) &&
```

Comment:
Do we still need to compare `isTrialSite` here? I thought a trial site would be marked as `isAtomic`.

Reply:
Yeah interesting, because it seems all trial sites would be atomic ... unless it's a trial site that's been reverted to simple? But in that case you wouldn't want to `renderUploadCard()` 🤔

It'll all be cleared away in follow up PRs anyway. I wanted to split the work into multiple PRs otherwise it got way too big. I've got sub-tasks in the parent issue, but it's something like first removing all the trial modals, then we can remove the query hooks used to decide whether to show the modals, and then we can remove the API endpoint called by the hook.

And that should mean we're sure that removing the endpoint won't inadvertently break any clients.